### PR TITLE
Split base classes from Fortran2003.py file

### PR DESCRIPTION
**Pull request by rupertford**
This is a relatively simple change that removes the base classes, walk function and exceptions from Fortran2003.py and places them in a utils.py file. The Fortran2003.py file now only contains the Fortran2003.py rules. I've not tried to make any code improvements as this is a simple separation of code.

### CODE REVIEW

File: src/fparser/two/utils.py [hunk 1]
```diff
@@ -0,0 +1,1216 @@
+#!/usr/bin/env python
```
Comment (arporter) [inline src/fparser/two/utils.py]: I think we only need this line if the file is to be executable as it tells bash how to interpret the script.

File: src/fparser/two/utils.py [hunk 2]
```diff
@@ -0,0 +1,1216 @@
...
+# ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+
+'''Base classes and exception handling for fortran parser
```
Comment (arporter) [inline src/fparser/two/utils.py]: "Fortran"

File: src/fparser/two/utils.py [hunk 3]
```diff
@@ -0,0 +1,1216 @@
...
+class NoMatchError(Exception):
+    pass
+
+
+class FortranSyntaxError(Exception):
+    '''An exception indicating that the fparser believes the provided code
+    to be invalid Fortran
+
+    '''
+    pass
```
Comment (arporter) [inline src/fparser/two/utils.py]: I don't think you need `pass` if the class has a docstring.

File: src/fparser/two/utils.py (hunk 1 above)
Comment (rupertford) [reply src/fparser/two/utils.py]: Removed.

File: src/fparser/two/utils.py (hunk 2 above)
Comment (rupertford) [reply src/fparser/two/utils.py]: You might as well just add this for every review - I always have at least one :-)

File: src/fparser/two/utils.py (hunk 3 above)
Comment (rupertford) [reply src/fparser/two/utils.py]: I didn't know that!

**Comment by rupertford** [comment]
Addressed reviewers comments (@arporter). Ready for 2nd review.

**Pull request merged**